contracts/oracle/MarketGovernanceOracle.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {PCVOracle} from "./PCVOracle.sol";
+import {CoreRefV2} from "./../refs/CoreRefV2.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 30.42 day period into the VOLT price
+/// after the oracle start time.
+/// interest can be compounded at any point in time if the Volt rate changes due to
+/// excessive illiquid composition of PCV.
+/// @author Elliot Friedman
+contract MarketGovernanceOracle is CoreRefV2 {
+    using SafeCast for *;
+
+    /// @notice event emitted when the Volt system oracle compounds
+    /// emits the end time of the period that completed and the new oracle price
+    event InterestCompounded(uint256 periodStart, uint256 newOraclePrice);
+
+    /// @notice event emitted when the Volt system oracle actual rate updates
+    event ActualRateUpdated(uint256 oldRate, uint256 newRate);
+
+    /// @notice event emitted when reference to the PCV Oracle updates
+    event PCVOracleUpdated(address oldOracle, address newOracle);
+
+    /// ---------- Mutable Variables ----------
+
+    /// @notice acts as an accumulator for interest earned in previous periods
+    /// returns the oracle price from the end of the last period
+    uint256 public oraclePrice;
+
+    /// @notice start time at which point interest will start accruing, and the
+    /// current ScalingPriceOracle price will be snapshotted and saved
+    uint256 public periodStartTime;
+
+    /// @notice current amount that oracle price is inflating by monthly as a percent, scaled by 1e18
+    /// set by governance, not actual rate Volt increases by
+    uint256 public baseChangeRate;
+
+    /// @notice actual amount that oracle price is inflating by monthly as a percent, scaled by 1e18
+    /// derived from base rate volt price interpolates by this amount.
+    uint256 public actualChangeRate;
+
+    /// @notice reference to the PCV oracle smart contract
+    address public pcvOracle;
+
+    /// ---------- Immutable Variable ----------
+
+    /// @notice the percentage at which the Volt rate will start going above the base rate
+    uint256 public immutable liquidityJumpTarget;
+
+    /// ---------- Constant Variables ----------
+
+    /// @notice maximum amount that oracle price can inflate by monthly as a
+    /// percentage, derived from base rate. 10% is max monthly price increase
+    uint256 public constant maximumChangeRate = 1e17;
+
+    /// @notice the time frame over which all changes in the APR are applied
+    /// one month was chosen because this is a temporary oracle
+    uint256 public constant TIMEFRAME = 30.42 days;
```

Comment:
why not make the changes happen over a year?

Reply:
If we want rate to "spike" to respond to changes in liquidity, we need a short timeframe (but not too short, to not create unnecessary MEV). Example :
- Current rate = 2%
- Allocation to illiquid venues increase, the rate model wants to change VOLT rate to 6%
- If TIMEFRAME = 30 days, it will take 1 month to go from 2% to 6%. I think it's a reasonable time, could even be shorter (1 week?)
- If TIMEFRAME = 1 year, the VOLT rate will be 2.3% after 1 month, barely any change, certainly not a large enough change to have opportunist deposits that increase liquidity for redemptions

Reply:
ha yes ok, good point, let's use a 1 year timeframe then because then the rate can be an APR and it's more intuitive